package com.lxjl.juling.module.mes.service.md.autocode.strategy;

import com.lxjl.juling.module.mes.dal.dataobject.md.autocode.MesMdAutoCodePartDO;

/**
 * MES 编码规则分段策略接口
 *
 * @author 棱信矩灵
 */
public interface MesMdAutoCodePartStrategy {

    /**
     * 获取策略类型
     *
     * @return 分段类型
     */
    Integer getType();

    /**
     * 生成分段内容
     *
     * @param part 规则组成
     * @param context 上下文
     * @return 分段内容
     */
    String generate(MesMdAutoCodePartDO part, MesMdAutoCodeContext context);

}
